File: bookhub-back/src/main/java/fr/eni/bookhub/entity/Livre.java
package fr.eni.bookhub.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;

@Entity
@Table(name = "livre")
@Data
public class Livre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "isbn", length = 13, nullable = false, unique = true)
    private String isbn;

    @Column(name = "titre", nullable = false, length = 255)
    private String titre;

    @Column(name = "resume", length = 255)
    private String resume;

    @Column(name = "image_couverture", length = 255)
    private String imageCouverture;

    @Column(name = "nb_page")
    private Integer nbPage;

    @Column(name = "date_parution")
    private LocalDate dateParution;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_auteur")
    private Auteur auteur;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_categorie")
    private Categorie categorie;
}

File: bookhub-front/src/app/core/modeles/livre.ts
import { type Auteur } from './auteur';
import { type Categorie } from './categorie';

export type Livre = {
  id: number,
  isbn: string,
  titre: string,
  resume: string,
  imageCouverture: string,
  nbPage: number,
  auteurId: number,
  auteurNom: string,
  auteurPrenom: string,
  categorieId: number,
  categorieLibelle: string,
  nbExemplaire: number,
  nbExemplaireDispo: number,
  note: number,
  dateParution: string,
}
